### ISSUE TITLE
SSO Auth

### ISSUE THREAD

**Issue by warriorsoul15** [shibdib/Dramiel#192]
So I have been using Dramiel for some time and decided to setup SSO Auth...

It appears to be working but the webpage after it returns from Eve SSO stays the same with the "log in with eve online" button but I notice the URL line changes to something like this:

https://www.xxxxxxxxxxxxxxxxxxxx.com/auth/?code=--somelongapicodehere

I am assuming that is the code... but its not presented in any kind of user friendly way..   did something go wrong?

**Comment by wahkawahka** [comment]
For what its worth - I am having this same issue. once I complete the CCP authentication I am sent back to my site I am at the same 'code=' url. with the "click the button below to login with your eve online account". 

Nothing is being posted to the SQL database - my pending table is empty. I tested the sql user credentials and was able to add data to the table so it does not appear to be a SQL account issue.

I have also tested the discord invite url - which is also working.

I am stepping through the code to try and determine what it may be stuck on. I did notice that the app.php file was looking for a /vendor/autoload.php file which I did not have. Looking into that further and re-running composer for the auth, I added the teamreflex/DiscordPHP api wrapper. which solved that problem.

However I am still stuck. I would appreciate any suggestions on how I might enable a higher level of logging to track the problem down. ty

**Comment by shibdib** [comment]
If I had to guess you setup your application wrong on the eve developers site.

If the link your bot gives you is http://discord.mambaonline.org _(Notice that it doesn't have /auth/)_

The callback URL should be http://discord.mambaonline.org/auth/

**Comment by wahkawahka** [comment]
Thank you. The callback url on the dev site is https://www.<mydomain>.com/auth/

I created a symlink on the root of my web site pointing 'auth' to the public folder of the sso auth application and I can access that successfully from https://www.<mydomain>.com/auth/ 

The URL that is returned after successful ccp logon is https://www.<mydomain>.com/auth/?code=kU9fxhnpjH9hc6nvNE2qXXXLgGSoTJCCbHbZp74Br05lj77CjUPTw-JrN8wiST0

**Comment by warriorsoul15** [comment]
Same for me

**Comment by wahkawahka** [comment]
*facepalm* thanks for the kick in the pants. 

the relationship between the root of the site and the /auth/ was lost on me in your example because I did not see a /auth/ directory and assumed that needed to be where the application lived (pointed at the public directory).

I am in the process of moving things around so that the public folder is presented in /discord/ on my site which would then send the callback from ccp to /discord/auth/.

**Comment by warriorsoul15** [comment]
wahka,

I am not following what you mean there.    I tried putting the core of the app in a   www.xxxx.com/discord   directory and making a symlink to   www.xxxx.com/discord/public   to   'auth' so   www.xxxx.com/discord/auth would work   and same issue.

**Comment by warriorsoul15** [comment]
Hmm I hashed through all that... now im getting codes but !auth with the code gives null response lol   ahhh it cant be in a private message!

nice!  its working now although I have not tried the persistent API checks

**Comment by wahkawahka** [comment]
On my server : Apache root is /var/www/html
my bot is installed in /var/www/dramiel
my sso auth is installed in /var/www/sso

So my symlink was named auth and located in the root of the web server (/var/www/html) pointing to /var/www/sso/public I changed the name of the link to discord and reversed a lot of other changes I made as I was fighting with the issue to get it to work.

I also changed the callback url to www.server.com/discord/auth/ 

Understand that /auth/ is not a directory or a file - but some kind of hook that triggers the code on your server to act once the SSO auth passes from CCP to your host.

Check the permissions of files and folders. Also check your .htaccess file in the /public folder. I think I needed to remove a / in mine as the .htaccess file contains rewrites for that folder which apply during the callback.

HTH
